code/Russell/lab9_1.py

```diff
@@ -0,0 +1,90 @@
+import json
+
+
+class ContactList:
+    
+    def __init__(self):
+        self.contacts = []
+
+    def load(self):
+
+        with open('contacts.json', 'r') as f: # 1) open 'contacts.json' with option 'r' for read
+            contents = f.read() # 2) get the text from the file
+            data = json.loads(contents) # 3) convert the text into a python dictionary (json.loads)
+        for contact in data['contacts']: # 4) get the list of contacts out of the dictionary
+            self.contacts.append(contact) # 5) assign the list of dictionaries to self.contacts
+        
+       
+    
+    def count(self):
+        return len(self.contacts) # return the length of self.contacts
+        
+    
+    def save(self):
+        with open('contacts.json', 'w') as f: # 1) open 'contacts.json' with open 'w' for write
+            contact_dict = {'contacts':self.contacts} # 2) put self.contacts in a dictionary with the key 'contacts'
+            json_string = json.dumps(contact_dict) # 3) convert the dictionary to a json string (json.dumps)
+            f.write(json_string) # 4) write the json string to the file
+
+    def print(self):
+        for contact in self.contacts: # loop over self.contacts
+            print(contact) # print the information for each contact on a separate line
+
+    def add(self, name, phone_number, email):
+        self.new_contact = {'name': name, 'phone_number':phone_number, 'email':email} # create a new dictionary using the 3 parameters
+        self.contacts.append(self.new_contact)# add the new dictionary to self.contacts
+    
+    def remove(self, name):
+        for contact in self.contacts: # find the contact in self-contacts with the given name
+            if contact['name'] == name:
+                self.contacts.remove(contact) # remove the element at that index
+    
+    def update(self, old_name, new_name, new_phone_number, new_email):
+        for contact in self.contacts: # find the contact in self.contacts with the given old_name
+            if contact['name'] == old_name:
+                ContactList.add(self, new_name, new_phone_number, new_email)
```

Comment:
Here we are getting the same result as if we used the add function on line 33. Instead of adding, we want to replace the values for that contact in the dictionary. 

let me know if you need help.